### PR TITLE
Added dockerfile for faucet.

### PR DESCRIPTION
**Pull request by arhamchordia**
Used node:slim image instead of node:alpine, as it is smaller in size .

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
+FROM node:slim as build-env
+
+WORKDIR /home/faucet/
+COPY . .
+
+RUN npm install
```
Comment (puneet2019) [inline Dockerfile:6]: `npm ci`

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+FROM alpine:3.15
+# Installs Node.js 16 (https://pkgs.alpinelinux.org/packages?name=nodejs&branch=v3.15)
+RUN apk add --update nodejs
+RUN apk add --update ca-certificates jq bash curl
+
+WORKDIR /home/faucet
+
+COPY . .
+
+EXPOSE 5001
```
Comment (puneet2019) [inline Dockerfile:19]: why expose 5001?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+# Use Alpine and install Node.js which is 50% smaller than the -alpine version of the node
+# image (53 MB including the faucet app).
+FROM alpine:3.15
+# Installs Node.js 16 (https://pkgs.alpinelinux.org/packages?name=nodejs&branch=v3.15)
+RUN apk add --update nodejs
+RUN apk add --update ca-certificates jq bash curl
+
+WORKDIR /home/faucet
+
+COPY . .
```
Comment (Anmol1696) [inline Dockerfile:17]: why copy the code?
No need to copy the bundle itself?
What is the benefit of running npm build in the above build docker env then? is it just a check?

Comment (Anmol1696) [reply Dockerfile:17]: If that is the case, can you check the size of the node:slim image and alpine image with dependencies installed.
Ideally we do this to reduce the size of the final image, but if node:slim itself is ok comparable size, then 2 images are not needed

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,20 @@
+FROM node:slim as build-env
```
Comment (Anmol1696) [inline Dockerfile:1]: with dockerfile we also should be adding in commands in Makefile to build and push the dockerfile with specific tags

**Review by Anmol1696 — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 3 above)
Comment (puneet2019) [reply Dockerfile:17]: he is doing npm install actually, yea. we should do build. it is better, then have the dist moved

Comment (vishrutsingh) [reply Dockerfile:17]: npm build is an internal function already called by npm install, but there is nothing to build its a js which can just be run right ?

File: Dockerfile (hunk 2 above)
Comment (arhamchordia) [reply Dockerfile:19]: It was a specific need for devote. 
Not needed anymore

File: Dockerfile (hunk 3 above)
Comment (arhamchordia) [reply Dockerfile:17]: Size with both is around ~150MB and just with slim is ~250MB.

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+# Installs Node.js 16 (https://pkgs.alpinelinux.org/packages?name=nodejs&branch=v3.15)
+
+RUN apk add --update nodejs
+RUN apk add --update npm
+RUN apk add --update ca-certificates jq bash curl
+
+WORKDIR /home/faucet/
+COPY . .
+
+RUN npm install
```
Comment (puneet2019) [inline Dockerfile:13]:
```suggestion
RUN npm ci
```

**Comment by snj33v** [comment]
bundle it and create new image from it
`npx esbuild server.js --bundle  --platform=node --outfile=index.js --tree-shaking=true`

**Review by snj33v — APPROVED** [review]

**Review by Anmol1696 — CHANGES_REQUESTED** [review]
add makefile

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,19 @@
+FROM node:slim as build-env
+
```
Comment (Anmol1696) [inline Dockerfile:2]: why extra line?

File: Dockerfile [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+WORKDIR /faucet/
+COPY . .
+
+RUN npm ci
+RUN npx esbuild server.js --bundle --platform=node --outfile=index.js
+
+
```
Comment (Anmol1696) [inline Dockerfile:10]: extra line

**Review by Anmol1696 — DISMISSED** [review]

**Review by puneet2019 — APPROVED** [review]

**Review by snj33v — APPROVED** [review]

**Pull request merged**